IdentityCore/src/webview/embeddedWebview/challangeHandlers/ios/MSIDCertAuthHandler.m

```diff
@@ -138,74 +92,26 @@ + (BOOL)handleChallenge:(NSURLAuthenticationChallenge *)challenge
-    redirectURI = newQueryItems[MSID_OAUTH2_REDIRECT_URI];
     
-    s_systemWebViewController = nil;
-    s_certAuthInProgress = YES;
-    
-    [MSIDMainThreadUtil executeOnMainThreadIfNeeded:^{
-        // This will launch a Safari view within the current Application, removing the app flip. Our control of this
-        // view is extremely limited. Safari is still running in a separate sandbox almost completely isolated from us.
-        
-        NSURL *currentURL = requestURL;
-        
-        if (s_useLastRequestURL && webview.URL)
-        {
-            currentURL = webview.URL;
-        }
-        
-        s_systemWebViewController = [[MSIDSystemWebviewController alloc] initWithStartURL:currentURL
-                                                                              redirectURI:redirectURI
-                                                                         parentController:parentViewController
-                                                                 useAuthenticationSession:s_useAuthSession
-                                                                allowSafariViewController:YES
-                                                               ephemeralWebBrowserSession:YES
-                                                                                  context:context];
-        
-        s_systemWebViewController.appActivities = s_activities;
+    [MSIDCertAuthManager.sharedInstance startWithURL:requestURL
+                                    parentController:parentViewController
+                                             context:context
+                                     completionBlock:^(NSURL *callbackURL, NSError *error)
+     {
+        MSIDWebviewSession *session = [MSIDWebviewAuthorization currentSession];
+        MSIDOAuth2EmbeddedWebviewController *embeddedViewController = (MSIDOAuth2EmbeddedWebviewController  *)session.webviewController;
         
-        [s_systemWebViewController startWithCompletionHandler:^(NSURL *callbackURL, NSError *error) {
+        [MSIDMainThreadUtil executeOnMainThreadIfNeeded:^{
             
-            MSIDWebviewSession *session = [MSIDWebviewAuthorization currentSession];
-            MSIDOAuth2EmbeddedWebviewController *embeddedViewController = (MSIDOAuth2EmbeddedWebviewController  *)session.webviewController;
-            
-            [MSIDMainThreadUtil executeOnMainThreadIfNeeded:^{
-                
-                if (callbackURL || error)
-                {
-                    [embeddedViewController endWebAuthWithURL:callbackURL error:error];
-                }
-                else
-                {
-                    NSError* unexpectedError = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, @"Unexpected Cert Auth response received.", nil, nil, nil, nil, nil, YES);
-                    [embeddedViewController endWebAuthWithURL:nil error:unexpectedError];
-                }
-            }];
+            if (callbackURL || error)
```

Comment:
Is this new DUNA callback? If yes, how are we maintaining previous logic in case we need to roll back to it?

Reply:
No it original logic, see [here](https://github.com/AzureAD/microsoft-authentication-library-common-for-objc/blob/a00e8b5fdb75ad83663c3cffce61509e33043a4e/IdentityCore/src/webview/embeddedWebview/challangeHandlers/ios/MSIDCertAuthHandler.m#L199).

No new logic was added to this class, only some logic was moved to MSIDCertAuthManager.